From 7a6289c5ace52cf88f0e19caa5f78b7c15d0e7a6 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@collabora.co.uk>
Date: Wed, 21 Jul 2021 17:42:30 +0000
Subject: [PATCH] fxcodec, fxge: fix missing includes with libstdc++

These missing includes break the build with gcc/libstdc++, they were not
a problem in practice with clang/libc++.

Change-Id: I40013f97ba7ab06f32aa59f87b04aec06a19478c
Reviewed-on: https://pdfium-review.googlesource.com/c/pdfium/+/83210
Commit-Queue: Lei Zhang <thestig@chromium.org>
Reviewed-by: Lei Zhang <thestig@chromium.org>
---

diff --git a/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.cpp b/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.cpp
index cea0679..036f250 100644
--- a/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.cpp
+++ b/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.cpp
@@ -7,6 +7,7 @@
 #include "core/fxcodec/jpeg/jpegmodule.h"
 
 #include <setjmp.h>
+#include <string.h>
 
 #include <memory>
 #include <utility>
diff --git a/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.cpp b/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.cpp
index c66985a..9c1122b 100644
--- a/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.cpp
+++ b/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxcodec/jpx/cjpx_decoder.h"
 
+#include <string.h>
+
 #include <algorithm>
 #include <limits>
 #include <utility>
diff --git a/third_party/pdfium/core/fxge/cfx_cliprgn.cpp b/third_party/pdfium/core/fxge/cfx_cliprgn.cpp
index 5369d52..d198852 100644
--- a/third_party/pdfium/core/fxge/cfx_cliprgn.cpp
+++ b/third_party/pdfium/core/fxge/cfx_cliprgn.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/cfx_cliprgn.h"
 
+#include <string.h>
+
 #include <utility>
 
 #include "core/fxge/dib/cfx_dibitmap.h"
diff --git a/third_party/pdfium/core/fxge/dib/cfx_bitmapcomposer.cpp b/third_party/pdfium/core/fxge/dib/cfx_bitmapcomposer.cpp
index 6f9b420..0f1ffae 100644
--- a/third_party/pdfium/core/fxge/dib/cfx_bitmapcomposer.cpp
+++ b/third_party/pdfium/core/fxge/dib/cfx_bitmapcomposer.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/dib/cfx_bitmapcomposer.h"
 
+#include <string.h>
+
 #include "core/fxge/cfx_cliprgn.h"
 #include "core/fxge/dib/cfx_dibitmap.h"
 
diff --git a/third_party/pdfium/core/fxge/dib/cfx_bitmapstorer.cpp b/third_party/pdfium/core/fxge/dib/cfx_bitmapstorer.cpp
index f57c00e..45a0a18 100644
--- a/third_party/pdfium/core/fxge/dib/cfx_bitmapstorer.cpp
+++ b/third_party/pdfium/core/fxge/dib/cfx_bitmapstorer.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/dib/cfx_bitmapstorer.h"
 
+#include <string.h>
+
 #include <utility>
 
 #include "core/fxge/dib/cfx_dibitmap.h"
diff --git a/third_party/pdfium/core/fxge/dib/cfx_dibbase.cpp b/third_party/pdfium/core/fxge/dib/cfx_dibbase.cpp
index 4ec0ddb..a1de2fb 100644
--- a/third_party/pdfium/core/fxge/dib/cfx_dibbase.cpp
+++ b/third_party/pdfium/core/fxge/dib/cfx_dibbase.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/dib/cfx_dibbase.h"
 
+#include <string.h>
+
 #include <algorithm>
 #include <memory>
 #include <utility>
diff --git a/third_party/pdfium/core/fxge/dib/cfx_dibitmap.cpp b/third_party/pdfium/core/fxge/dib/cfx_dibitmap.cpp
index d7ccf6c..94e8acc 100644
--- a/third_party/pdfium/core/fxge/dib/cfx_dibitmap.cpp
+++ b/third_party/pdfium/core/fxge/dib/cfx_dibitmap.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/dib/cfx_dibitmap.h"
 
+#include <string.h>
+
 #include <limits>
 #include <memory>
 #include <utility>
diff --git a/third_party/pdfium/core/fxge/dib/cfx_scanlinecompositor.cpp b/third_party/pdfium/core/fxge/dib/cfx_scanlinecompositor.cpp
index e8362d7..c04c6dc 100644
--- a/third_party/pdfium/core/fxge/dib/cfx_scanlinecompositor.cpp
+++ b/third_party/pdfium/core/fxge/dib/cfx_scanlinecompositor.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxge/dib/cfx_scanlinecompositor.h"
 
+#include <string.h>
+
 #include <algorithm>
 
 #include "core/fxge/dib/fx_dib.h"
